pkg/provisioner/ironic/ironic.go

```diff
@@ -622,6 +622,28 @@ func (p *ironicProvisioner) getUpdateOptsForNode(ironicNode *nodes.Node) (update
 		},
 	)
 
+	// image_checksum
+	//
+	// FIXME: For older versions of ironic that do not have
+	// https://review.opendev.org/#/c/711816/ failing to include the
+	// 'image_checksum' causes ironic to refuse to provision the
+	// image, even if the other hash value parameters are given.
+	if _, ok := ironicNode.InstanceInfo["image_checksum"]; !ok {
```

Comment:
Should we condition this on the algorithm being MD5? If it's a different type then we will end up reporting that the checksum is wrong when in fact we just can't support it at all (on older versions of Ironic).

Reply:
Done